### PR TITLE
fix: `useEventBus` by propagating config to `DefaultEventsService`

### PR DESCRIPTION
**Pull request by RedlineTriad**
## Hey, I just made a Pull Request!

This setting was added by reading from an optional `config` argument. This was never passed, so `auto` was always assumed. https://github.com/backstage/backstage/pull/27227/files#diff-cffd40a187b4fc584f03c968517fbacc110bde6a467f2384bbd92b803d3db47eR374-R379

I noticed this issue because when using the proxy, event service has serious issues, and even seems to break login somehow?
(I think it uses a local `172.*.*.*` IP instead of `localhost` or `127.0.0.1`, and it can't parse IP ranges in `GLOBAL_AGENT_NO_PROXY`)
But I will hunt that issue in a different PR.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by freben** [comment]
Hi! Note that you'll need to run `yarn build:api-reports` when API surfaces change.

**Comment by RedlineTriad** [comment]
> Hi! Note that you'll need to run `yarn build:api-reports` when API surfaces change.

I know, I forget it every time...

File: plugins/events-backend/report.api.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ import { LoggerService } from '@backstage/backend-plugin-api';
 // @public @deprecated
 export class DefaultEventBroker implements EventBroker {
   // @deprecated
-  constructor(logger: LoggerService, events?: EventsService);
+  constructor(logger: LoggerService, config?: Config, events?: EventsService);
```
Comment (Rugvip) [inline plugins/events-backend/report.api.md:21]: Can't break the API like this. Let's leave the config out of this one and instead hardcode it to `'never'` for the old event broker implementation.

File: plugins/events-backend/report.api.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ export class DefaultEventBroker implements EventBroker {
 
 // @public @deprecated
 export class EventsBackend {
-  constructor(logger: Logger);
+  constructor(logger: Logger, config?: Config);
```
Comment (Rugvip) [inline plugins/events-backend/report.api.md:32]: Same here, can skip this change

**Review by Rugvip — COMMENTED** [review]
Thank you! 👍 

Just need some tweaks, although I'll get on those so we can ship this asap

**Review by Rugvip — APPROVED** [review]
Alright, thank you! Gonna :shipit:

**Pull request merged**